src/ml/neural_net/image_augmentation.cpp

```diff
@@ -35,6 +35,11 @@ shared_float_array convert_to_shared_float_array(
 
 std::vector<image_annotation> convert_to_image_annotation(
     shared_float_array augmented_annotation) {
+  // Check if the annotation is empty
+  if (augmented_annotation.size() == 1) {
```

Comment:
Please check for `.dim()` being 0 instead. Size equal 1 doesn’t read empty in the same way

Reply:
it seems like empty annotation is not covered by our python unit test.

Reply:
@Jarvi-Izana I think it is covered:
```
for j in range(rs.randint(max_num_boxes_per_image)):
            left, right = np.sort(rs.randint(0, img_shape[1], size=2))
            top, bottom = np.sort(rs.randint(0, img_shape[0], size=2))
```
the bounding box number is generated randomly,
so that why I found this bug ;) .

Reply:
lol, good